reporter/html.js

```diff
@@ -206,7 +206,13 @@ export function escapeText( s ) {
 			config[ field.name ] = value || false;
 			tests = id( "qunit-tests" );
 			if ( tests ) {
-				toggleClass( tests, "hidepass", value || false );
+				var children = tests.children;
+				for ( var i = 0; i < children.length; i++ ) {
+					var test = children[ i ];
+					if ( test && test.className.indexOf( "pass" ) > -1 ) {
+						tests.removeChild( test );
```

Comment:
Yes please. If we retain the nodes in memory, will that cause issues with a test suite of the size y'all are working with?

Reply:
The only way I can see this working is if we 

1. retain all the nodes in memory and basically filter what is displaying based on the checkbox value. If it is checked, we update the list with only the failed, skipped or todo tests, if it isn't we display all. 

2. have a function that creates the output items using the data in memory so that we don't have to keep a reference to the dom element, but only to the data, then create the list using a function that uses the data to make the dom element.

Reply:
I had the same thoughts. Both approaches seem reasonable to me, but retaining the nodes would be predicated on it not causing too much memory pressure for large test suites such as the one y'all have. I don't think it should be an issue though.

Basically, what I'm saying is choice is up to you if both approaches work :)

Reply:
Sorry for the late response, pushed code that will keep hiddenTests in memory and not disrupt the current behavior.